lib/cmds/update.ts

```diff
@@ -118,63 +128,74 @@ function update(options: Options): void {
-    });
+    updateBrowserFile(binary, outputDir);
+    promises.push(
+        FileManager.downloadFile(binary, outputDir, proxy, ignoreSSL)
+            .then<void>((downloaded: boolean) => {
+              if (!downloaded) {
+                logger.info(
+                    binary.name + ': file exists ' +
+                    path.resolve(outputDir, binary.filename(Config.osType(), Config.osArch())));
+                logger.info(binary.name + ': ' + binary.versionCustom + ' up to date');
+              }
+            }));
   }
   if (chrome) {
     let binary = binaries[ChromeDriver.id];
-    updateBinary(binary, outputDir, proxy, ignoreSSL);
+    updateBrowserFile(binary, outputDir);
+    promises.push(updateBinary(binary, outputDir, proxy, ignoreSSL));
   }
   if (gecko) {
     let binary = binaries[GeckoDriver.id];
-    updateBinary(binary, outputDir, proxy, ignoreSSL);
+    updateBrowserFile(binary, outputDir);
+    promises.push(updateBinary(binary, outputDir, proxy, ignoreSSL));
   }
   if (ie) {
     let binary = binaries[IEDriver.id];
     binary.arch = Config.osArch();  // Win32 or x64
-    updateBinary(binary, outputDir, proxy, ignoreSSL);
+    updateBrowserFile(binary, outputDir);
+    promises.push(updateBinary(binary, outputDir, proxy, ignoreSSL));
   }
   if (ie32) {
     let binary = binaries[IEDriver.id];
     binary.arch = 'Win32';
-    updateBinary(binary, outputDir, proxy, ignoreSSL);
+    updateBrowserFile(binary, outputDir);
+    promises.push(updateBinary(binary, outputDir, proxy, ignoreSSL));
   }
   if (android) {
     let binary = binaries[AndroidSDK.id];
     let sdk_path = path.resolve(outputDir, binary.executableFilename(Config.osType()));
     let oldAVDList: string;
 
-    q.nfcall(fs.readFile, path.resolve(sdk_path, 'available_avds.json'))
-        .then(
-            (oldAVDs: string) => {
-              oldAVDList = oldAVDs;
-            },
-            () => {
-              oldAVDList = '[]';
-            })
-        .then(() => {
-          return updateBinary(binary, outputDir, proxy, ignoreSSL);
-        })
-        .then(() => {
-          initializeAndroid(
-              path.resolve(outputDir, binary.executableFilename(Config.osType())),
-              android_api_levels, android_architectures, android_platforms, android_accept_licenses,
-              binaries[AndroidSDK.id].versionCustom, JSON.parse(oldAVDList), logger, verbose);
-        })
-        .done();
+    promises.push(q.nfcall(fs.readFile, path.resolve(sdk_path, 'available_avds.json'))
+                      .then(
+                          (oldAVDs: string) => {
+                            oldAVDList = oldAVDs;
+                          },
+                          () => {
+                            oldAVDList = '[]';
+                          })
+                      .then(() => {
+                        return updateBinary(binary, outputDir, proxy, ignoreSSL);
+                      })
+                      .then(() => {
```

Comment:
Use `then<void>` to make this a `q.Promise<void>` instead of `q.Promise<string>`.